### ISSUE TITLE
Support Carthage

### ISSUE THREAD

**Issue by friedbunny** [mapbox/MapboxDirections.swift#46]
v0.5.0 does not support installation via Carthage. Here are the results when Carthage tries to build MapboxDirections.swift:

``` sh
$ carthage update
*** Fetching MapboxDirections.swift
*** Downloading MapboxDirections.swift.framework binary at "v0.5.0"
*** Checking out MapboxDirections.swift at "v0.5.0"
*** xcodebuild output can be found in /var/folders/s5/2p4466vn2cv5mslx9ydthw6m0000gn/T/carthage-xcodebuild.aYLxWu.log
*** Building scheme "MapboxDirections" in MapboxDirections.xcodeproj
** BUILD FAILED **

The following build commands failed:
    PhaseScriptExecution Check\ Pods\ Manifest.lock /Users/jason/Library/Developer/Xcode/DerivedData/MapboxDirections-awevmnvtblxkbgdzugqxbaoxvrlk/Build/Intermediates/MapboxDirections.build/Release-iphoneos/MapboxDirections.build/Script-E4F58AFE16C929A75F6A3D5E.sh
(1 failure)
error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
A shell task (/usr/bin/xcrun xcodebuild -project /Users/jason/Desktop/Carthage/Carthage/Checkouts/MapboxDirections.swift/MapboxDirections.xcodeproj -scheme MapboxDirections -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** BUILD FAILED **

The following build commands failed:
    PhaseScriptExecution Check\ Pods\ Manifest.lock /Users/jason/Library/Developer/Xcode/DerivedData/MapboxDirections-awevmnvtblxkbgdzugqxbaoxvrlk/Build/Intermediates/MapboxDirections.build/Release-iphoneos/MapboxDirections.build/Script-E4F58AFE16C929A75F6A3D5E.sh
(1 failure)
```

Cartfile:

``` ruby
github "Mapbox/MapboxDirections.swift"
```

/cc @1ec5

**Comment by 1ec5** [comment]
I think we should tackle #28 first. If we use Carthage instead of CocoaPods to manage the project and its dependencies, we can’t get into this situation.

**Comment by Eldorado234** [comment]
#102 solves this issue (for `swift3` only so far)

**Comment by 1ec5** [comment]
Fixed in #102.
